controller/konnect/ops/ops.go

```diff
@@ -282,6 +330,22 @@ func Update[
 		return ctrl.Result{}, fmt.Errorf("unsupported entity type %T", ent)
 	}
 
+	if err != nil {
+		if id != "" {
```

Comment:
Ditto regarding the convention of not returning a valid object when an error is not nil.